server/lib/manifest.py

```diff
@@ -0,0 +1,389 @@
+import os
+
+from girder.utility.model_importer import ModelImporter
+
+
+class Manifest:
+    """
+    Class that represents the manifest file.
+    Methods that add information to the manifest file have the form
+    add_<someProperty>
+    while methods that create chunks of the manifest have the form
+    create<someProperty>
+    """
+
+    def __init__(self, tale_license, item_ids=None):
+        """
+        Initialize the manifest document with base variables
+        :param license:
+        :param item_ids:
+        """
+        self.manifest = dict()
+        self.item_ids = item_ids
+        # Holds the SPDX of the license
+        self.license = tale_license
+        # Create a set that represents any external data packages
+        self.datasets = set()
+
+        self.folderModel = ModelImporter.model('folder')
+        self.itemModel = ModelImporter.model('item')
+        self.userModel = ModelImporter.model('user')
+
+    def generate_manifest(self, user, tale):
+        self.manifest.update(self.create_context())
+        self.manifest.update(self.create_basic_attributes(tale))
+        self.add_tale_creator(tale, user)
+        self.manifest['aggregates'] = []
+
+        if self.item_ids:
+            self.add_item_records(user, tale)
+        else:
+            self.add_tale_records(tale, user)
+        # Add any external datasets to the manifest
+        self.add_dataset_records(user)
+        self.add_system_files()
+
+    publishers = {
+        "DataONE":
+            {
+                "@id": "https://www.dataone.org/",
+                "@type": "Organization",
+                "legalName": "DataONE",
+                "Description": "A federated data network allowing access to science data"
+            },
+        "Globus":
+            {
+                "@id": "https://www.materialsdatafacility.org/",
+                "@type": "Organization",
+                "legalName": "Materials Data Facility",
+                "Description": "A simple way to publish, discover, and access materials datasets"
+            }
+    }
+
+    def create_basic_attributes(self, tale):
```

Comment:
One of the places where it'd be nice to have `self.tale`

Reply:
Done in 651f7a4